types/stylelint/index.d.ts

```diff
@@ -814,6 +814,38 @@ declare namespace stylelint {
 		column: number;
 	};
 
+	export type ProblemLocation =
+		| {
+				/**
+				 * The inclusive start position of the problem, relative to the
+				 * node's source text. If provided, this will be used instead of
+				 * `index`.
+				 */
+				start: Position;
+				/**
+				 * The exclusive end position of the problem, relative to the
+				 * node's source text. If provided, this will be used instead of
+				 * `endIndex`.
+				 */
+				end: Position;
+		  }
+		| {
+				/**
+				 * The inclusive start index of the problem, relative to the node's
+				 * source text.
+				 */
+				index: number;
+				/**
+				 * The exclusive end index of the problem, relative to the node's
+				 * source text.
+				 */
+				endIndex: number;
+		  }
+		| {
+				word: string;
+		  }
+		| {};
```

Comment:
Why did you choose not to add
```
| {
	index: 0;
	endIndex?: number;
}
```
?

Reply:
I've answered this here to keep the design discussion in one place: https://github.com/stylelint/stylelint/issues/8126#issuecomment-2559300114